HyperVExtension/src/HyperVExtension/CommunicationWithGuest/WmiUtility.cs

```diff
@@ -124,4 +138,47 @@ public static bool JobCompleted(ManagementBaseObject outParams, ManagementScope
 
         return jobCompleted;
     }
+
+    public static FeatureAvailabilityKind GetHyperVFeatureAvailability()
+    {
+        try
+        {
+            var searcher = new ManagementObjectSearcher($"SELECT InstallState FROM Win32_OptionalFeature WHERE Name = 'Microsoft-Hyper-V'");
+            var collection = searcher.Get();
+
+            foreach (var instance in collection)
+            {
+                if (instance?.GetPropertyValue("InstallState") is uint enablementState)

```

Comment:
Can instance be null if enumerating through the collection?

Reply:
Isn't there a null check there already?

Reply:
Not sure I understand the question but the compiler shows that instance can be a `ManagementObjectCollection?` value so I added the nullability check